test/torch/test_hook.py

```diff
@@ -11,194 +11,227 @@
-    def test_functional_same_in_both_imports(self, attr):
-        """This function tests that the hook modifies the behavior of
-        torch.nn.function regardless of the import namespace
-        """
-        self.setUp()
-        fattr = getattr(F, attr)
-        tattr = getattr(torch.nn.functional, attr)
-        x = torch.Tensor([1, -1, 3, 4])
-        assert (fattr(x) == tattr(x)).all()
-
-    def test_hook_tensor(self):
-        self.setUp()
-        x = torch.tensor([1.0, -1.0, 3.0, 4.0], requires_grad=True)
-        x.send(self.bob)
-        x = torch.tensor([1.0, -1.0, 3.0, 4.0], requires_grad=True)[0:2]
-        x.send(self.bob)
-
-    def test_properties(self):
-        self.setUp()
-        x = torch.Tensor([1, -1, 3, 4])
-        assert x.is_wrapper is False
-
-    def test_signature_cache_change(self):
-        """Tests that calls to the same method using a different
-        signature works correctly. We cache signatures in the
-        hook.build_hook_args_function dictionary but sometimes they
-        are incorrect if we use the same method with different
-        parameter types. So, we need to test to make sure that
-        this cache missing fails gracefully. This test tests
-        that for the .div(tensor) .div(int) method."""
-
-        self.setUp()
-
-        x = torch.Tensor([1, 2, 3])
-        y = torch.Tensor([1, 2, 3])
-
-        z = x.div(y)
-        z = x.div(2)
-        z = x.div(y)
-
-        assert True
-
-    def test_parameter_hooking(self):
-        """Test custom nn.Module and parameter auto listing in m.parameters()"""
-        self.setUp()
-
-        class MyLayer(torch.nn.Module):
-            def __init__(self):
-                super().__init__()
-                self.some_params = torch.nn.Parameter(torch.tensor([5.0]))
-
-        m = MyLayer()
-        out = list(m.parameters())
-        assert len(out) == 1
-        assert out[0] == m.some_params
-
-    def test_torch_module_hook(self):
-        """Tests sending and getting back torch nn module like nn.Linear"""
-        self.setUp()
-        model = nn.Linear(2, 1)
-        model_ptr = model.send(self.bob)
-        res = model_ptr.get()
+def test__init__():
+    hook = syft.TorchHook(torch, verbose=True)
+    assert torch.torch_hooked
+    assert hook.torch.__version__ == torch.__version__
+
+
+def test_torch_attributes():
+    with pytest.raises(RuntimeError):
+        syft.torch._command_guard("false_command", "torch_modules")
+
+    assert syft.torch._is_command_valid_guard("torch.add", "torch_modules")
+    assert not syft.torch._is_command_valid_guard("false_command", "torch_modules")
+
+    syft.torch._command_guard("torch.add", "torch_modules", get_native=False)
+
+
+def test_worker_registration():
+    hook = syft.TorchHook(torch, verbose=True)
+
+    me = hook.local_worker
+    boris = syft.VirtualWorker(id="boris", hook=hook, is_client_worker=False)
+    me.add_workers([boris])
+
```

Comment:
I'm not sure this is what we want to do: this involves massive duplicate code that self.setUp() did avoid previously.

Reply:
@robert-wagner what code design where you thinking of?

Reply:
@LaRiffle we don't need to call hook, we should be passing as arguments to the tests the fixtures in conftest as needed

Reply:
Perfect that works for me then :)

Reply:
The general idea was to reduce code duplicate across the project by moving the functionality of setup to a common fixture